### PR TITLE
[ERE-2174] Disable submit button after click

### PR DESCRIPTION
**Pull request by ScriptSmith**

### CODE REVIEW

**Comment by ppettitau** [comment]
I know this is WIP, but I'm wondering why there is javascript to submit the form in the first place? A button with a `type` of `submit` should submit the form automatically when clicked without requiring JS, so is there some other factor at play here?

**Comment by ScriptSmith** [comment]
@ppettitau I'll see if it can be done in html. My guess is that there are some complications due to form validation.

**Comment by ppettitau** [comment]
@ScriptSmith , you can hook validation logic into the form submit action rather than the submit button click.

**Comment by ScriptSmith** [comment]
@ppettitau 

I believe it's failing because browsers won't let you submit a form that's missing a required field:

https://jsfiddle.net/bgu9d0pn/

Unless you use JavaScript to submit like we do, in which case it will allow you to submit without requiring fields:

https://jsfiddle.net/09hqnwk8/1/

I think it would be worth using the browsers' built-in validation, so I might change how some of the tests work to support this.

**Comment by ppettitau** [comment]
> @ppettitau 
> 
> I believe it's failing because browsers won't let you submit a form that's missing a required field:
> 
> https://jsfiddle.net/bgu9d0pn/
> 
> Unless you use JavaScript to submit like we do, in which case it will allow you to submit without requiring fields:
> 
> https://jsfiddle.net/09hqnwk8/1/
> 
> I think it would be worth using the browsers' built-in validation, so I might change how some of the tests work to support this.
> 

Yes, that's right. Client side validation gives quicker feedback on simple field validation than server side, so if it's easy enough to get it working with our existing solution I think it's a positive.

**Comment by ScriptSmith** [comment]
Main issue I can see is that the `collapse` accordions won't expand to reveal the browser's error popup & the form's input when you save a form.

I've added the `novalidate` attribute to disable client validation for now.

I also attempted to use no `form` attribute in the submit buttons, but it doesn't work because it requires the form to be an ancestor of the button.

https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#implicit-submission
https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#association-of-controls-and-forms

**Comment by ppettitau** [comment]
> Main issue I can see is that the `collapse` accordions won't expand to reveal the browser's error popup & the form's input when you save a form.
> 
> I've added the `novalidate` attribute to disable client validation for now.
> 
> I also attempted to use no `form` attribute in the submit buttons, but it doesn't work because it requires the form to be an ancestor of the button.
> 
> https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#implicit-submission https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#association-of-controls-and-forms

I'm ok with the `novalidate` for now, considering this is the existing behaviour. Semantically (and in favour of accessibility), submit buttons _should_ be contained within a `form`, but understand you are just working with existing code so it is what it is.

File: rdrf/rdrf/templates/rdrf_cdes/clinician.html [hunk 1]
```diff
@@ -51,6 +51,7 @@
 
             $("#main-form").submit(function() {
                 $(window).unbind("beforeunload");
+                $("#submit-btn").prop("disabled", true);
```
Comment (ppettitau) [inline rdrf/rdrf/templates/rdrf_cdes/clinician.html:54]: Is this still required?

**Review by ppettitau — APPROVED** [review]
Looks good. 
This will stop a user clicking the Save button twice, but the form is still potentially open to being submitted multiple times by other means (enter button)? I haven't been able to reproduce it though, and it's not the likely cause of the original issue being addressed.

**Comment by ppettitau** [comment]
> Looks good. This will stop a user clicking the Save button twice, but the form is still potentially open to being submitted multiple times by other means (enter button)? I haven't been able to reproduce it though, and it's not the likely cause of the original issue being addressed.

Actually, looks like you can't submit the form (via enter key) when the submit button is disabled! :D

**Comment by ScriptSmith** [comment]
I didn't think to check double-enter, but it seems that clicking enter actually 'clicks' the button, as removing the `$("#submit-btn").prop("disabled", true);` line allows the form to be double-submitted again

**Review by Lexiel46 — APPROVED** [review]

**Pull request merged**